lib/modules/platform/gitea/index.ts

```diff
@@ -323,28 +323,35 @@ const platform: Platform = {
       throw new Error(REPOSITORY_BLOCKED);
     }
 
-    if (repo.allow_rebase && repo.default_merge_style === 'rebase') {
-      config.mergeMethod = 'rebase';
-    } else if (
-      repo.allow_rebase_explicit &&
-      repo.default_merge_style === 'rebase-merge'
-    ) {
-      config.mergeMethod = 'rebase-merge';
-    } else if (
-      repo.allow_squash_merge &&
-      repo.default_merge_style === 'squash'
-    ) {
-      config.mergeMethod = 'squash';
-    } else if (
-      repo.allow_merge_commits &&
-      repo.default_merge_style === 'merge'
-    ) {
-      config.mergeMethod = 'merge';
-    } else if (
-      repo.allow_fast_forward_only_merge &&
-      repo.default_merge_style === 'fast-forward-only'
-    ) {
-      config.mergeMethod = 'fast-forward-only';
+    const giteaOrdering: PRMergeMethod[] = [
+      'merge',
+      'rebase',
+      'rebase-merge',
+      'squash',
+      'fast-forward-only',
+    ];
+
+    const candidateMergeStyles = Array.from(
+      new Set<PRMergeMethod>([repo.default_merge_style, ...giteaOrdering]),
```

Comment:
What's the point of using a `Set` here?

- Less readable (you need to know that in JS sets have defined iteration order)
- Likely slower (to build up the set).